frontend/src/components/common/Filter/index.tsx

```diff
@@ -241,21 +236,6 @@ function filterCategoryValuesWithCount(
   return values.filter(({ count }) => count > 0);
 }
 
-/**
- * Returns true if category is not applicable.
- * @param categoryView
- * @returns true when category  is not applicable.
- */
-function isCategoryNA(categoryView: CategoryView): boolean {
```

Comment:
I moved the disabled checks from the filter component to the `useCategoryFilter` hook.